src/composables/usePool.ts

```diff
@@ -357,6 +357,30 @@ export function fiatValueOf(pool: Pool, shares: string): string {
   return bnum(shares).times(bptPriceFor(pool)).toString();
 }
 
+export function findTokenByAddress(pool: Pool, address: string) {
+  return pool.tokens.find(token => isSameAddress(token.address, address));
+}
+
+export function getUnderlyingTokens(pool: Pool, address: string) {
+  const token = findTokenByAddress(pool, address);
+
+  const underlyingTokens = token?.token.pool?.tokens || [];
+  return underlyingTokens.filter(
+    token => !includesAddress(pool.tokensList, token.address)
+  );
+}
+
+export function calculateTokenBPTShareByAddress(
+  pool: Pool,
+  address: string
+): string {
+  const token = findTokenByAddress(pool, address);
+  if (!token) return '0';
+  return bnum(token?.balance || '0')
+    .div(token.token.pool?.totalShares || 1)
```

Comment:
I think this is incorrect, this is getting the `totalShares` if the token itself is a pool. You instead need to find the parent pool of the current token.
```
- Pool 1
-- Token 1
-- Token 2 (pool 2)
---- Token 3
---- Token 4
```
Let's say you want the bpt share of Token 2. Your calculation is returning `totalShars` of Pool 2, but what you want is the `totalShares` of Pool 1 because that is the parent pool of Token 2.

Reply:
I changed this by passing `totalShares` as a prop in `TokenBreakdown` (recursive component)